src/VisualStudio/Core/Def/Implementation/AbstractEditorFactory.cs

```diff
@@ -361,5 +365,63 @@ private async Task FormatDocumentCreatedFromTemplateAsync(IVsHierarchy hierarchy
                 formattedText.Write(textWriter, cancellationToken: CancellationToken.None);
             });
         }
+
+        private static Project AddEditorConfigFiles(Project projectToAddTo, string projectFolder)
+        {
+            do
+            {
+                projectToAddTo = AddEditorConfigFile(projectToAddTo, projectFolder, out var foundRoot);
+
+                if (foundRoot)
+                {
+                    break;
+                }
+                projectFolder = Path.GetDirectoryName(projectFolder);
+            }
+            while (projectFolder is not null);
+
+            return projectToAddTo;
+
+            static Project AddEditorConfigFile(Project project, string folder, out bool foundRoot)
+            {
+                const string EditorConfigFileName = ".editorconfig";
+
+                foundRoot = false;
+
+                var editorConfigFile = Path.Combine(folder, EditorConfigFileName);
+
+                var text = IOUtilities.PerformIO(() =>
+                {
+                    using var stream = File.OpenRead(editorConfigFile);
+                    return SourceText.From(stream);
+                });
+
+                if (text is null)
+                {
+                    return project;
+                }
+
+                // We need to stop looking in the file system if the .editorconfig file is a root file
+                // The root property must be at the top of the file, barring comments and empty lines,
+                // so we can just do a simple parse. If this misses some odd cases we're okay with it as
+                // it's only used for the temporary project anyway.
+                foreach (var line in text.Lines)
+                {
+                    if (line.Span.Length == 0 ||
```

Comment:
instead of Length==0, should we check if it's blank and skip that instead?  i worry about a single line with just one space on it messing this up.

Reply:
Removed all of this